library/nats-streaming

```diff
@@ -1,6 +1,25 @@
 Maintainers: Ivan Kozlovic <ivan.kozlovic@apcera.com> (@kozlovic)
-
-Tags: 0.5.0, latest
 GitRepo: https://github.com/nats-io/nats-streaming-docker.git
-GitCommit: 3301b7b0e11bfd128eb4ba572fc1fdbfaab8d3e3
+GitCommit: 2f9abaf0830c64e40d22e3a8d4da47b96c666aef
+
+Tags: 0.5.0-linux, linux
+SharedTags: 0.5.0, latest
+amd64-GitCommit: 2f9abaf0830c64e40d22e3a8d4da47b96c666aef
+amd64-Directory: amd64
+arm32v7-GitCommit: 2f9abaf0830c64e40d22e3a8d4da47b96c666aef
+arm32v7-Directory: arm32v7
+arm64v8-GitCommit: 2f9abaf0830c64e40d22e3a8d4da47b96c666aef
+arm64v8-Directory: arm64v8
+
+Tags: 0.5.0-nanoserver, nanoserver
+SharedTags: 0.5.0, latest
+Architectures: windows-amd64
+windows-amd64-GitCommit: 2f9abaf0830c64e40d22e3a8d4da47b96c666aef
+windows-amd64-Directory: windows/nanoserver
+Constraints: nanoserver
 
+Tags: 0.5.0-windowsservercore, windowsservercore
+Architectures: windows-amd64
+windows-amd64-GitCommit: 2f9abaf0830c64e40d22e3a8d4da47b96c666aef
+windows-amd64-Directory: windows/windowsservercore
+Constraints: windowsservercore
```

Comment:
These images both appear to be using `ENTRYPOINT`, which isn't something we've allowed on Windows previously due to PowerShell or even CMD not being sufficient for an entrypoint script, lacking `exec` or even the ability to perform sane signal passing (see https://github.com/docker-library/official-images#consistency).

Reply:
I am confused about this one. Reading the link, I don't see why we would not be using entrypoint and cmd. The entrypoint is the nats-streaming-server executable and we pass by default the monitoring port (with cmd). Note this is also how we did for NATS Server.

Reply:
Looking at other repos for guidance. Should I remove entrypoint and have this for instance:
```
CMD ["nats-streaming-server", "-m", "8222"]
```
So if you run the docker image without any argument, it will launch the server with monitoring port. However, if one wants to change the parameters, one would have to include the executable `nats-streaming-server` as the first argument.
With that approach, though, one could start `cmd` and have a shell.

Reply:
Yeah, for now using `CMD` would definitely be the easiest path to acceptance -- we're trying to find a good solution for Windows to have `ENTRYPOINT` and still follow our guidelines, but we don't have any exceptions to the `ENTRYPOINT` rule except images that are `FROM scratch` (and thus have no environment to allow access to, which is why the existing images use `ENTRYPOINT`).

Reply:
Ok, so I will update the other repo and let you know when that is done. I am also checking with my team if we need to issue 0.6.2 with Go 1.9.2 or 0.6.0 with Go 1.9.1 is fine. Thanks!

Reply:
@tianon I have updated the Dockerfiles in the nats-streaming-docker repo, and updated the GitCommit only for the Windows variants, hope that is ok. Let me know. Thanks!